scene/gui/popup_menu.cpp

```diff
@@ -96,10 +96,21 @@ Size2 PopupMenu::_get_contents_minimum_size() const {
 		minsize.width += check_w;
 	}
 
+	return minsize;
+}
+
+Size2 PopupMenu::_get_contents_minimum_size() const {
+	Size2 minsize = get_contents_size();
+
 	if (is_inside_tree()) {
-		int height_limit = get_usable_parent_rect().size.height;
-		if (minsize.height > height_limit) {
-			minsize.height = height_limit;
+		if (get_height_limit() > -1) {
+			if (minsize.height > get_height_limit()) {
+				minsize.height = get_height_limit();
+			}
+		} else {
+			if (minsize.height > get_usable_parent_rect().size.height) {
+				minsize.height = get_usable_parent_rect().size.height;
+			}
```

Comment:
You should avoid calling `get_usable_parent_rect()` twice.
(and possible same above)

Or you could use `MIN()` too.

Reply:
@KoBeWi Thanks for the suggestion!